llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -1389,7 +1393,7 @@ AliasResult BasicAAResult::aliasPHI(const PHINode *PN, LocationSize PNSize,
     if (PV1 == PN)
       continue;
 
-    if (isa<PHINode>(PV1)) {
+    if (!EnableFullPHIAnalysis && isa<PHINode>(PV1)) {
```

Comment:
I feel ill at ease with not setting OnePhi, since this is used below. Would it make sense to just control the conservative return here?

Reply:
By not setting OnePhi, we will give the possibility to run some more pessimistic step further.

See:

```
  if (OnePhi && UniqueSrc.size() > 1)
    // Out of an abundance of caution, allow only the trivial lcssa and
    // recursive phi cases.
    return AliasResult::MayAlias;
```

If I just control the return, I loose some opportunities....

Reply:
Not sure I understand the surrounding code properly, but does that mean that for a diamond cfg, the phi node would not be followed because it has two distinct (non recursive) sources? Would it be possible to add a small `llc` test where we pass the `--basic-aa-full-phi-analysis` option?

Reply:
Hi @gbossu, I extended the existing test to accommodate this change.

Reply:
Hi @gbossu, I discover that a more complete phi analysis support was dropped in the past: https://reviews.llvm.org/D139719